gprofiler/profilers/python.py

```diff
@@ -170,11 +170,14 @@ def __init__(
         stop_event: Optional[Event],
         storage_dir: str,
         insert_dso_name: bool,
+        profiling_mode: str,
```

Comment:
I'm thinking maybe we need something like "traits" - instead of forcing all profilers to follow the API, we could define a "trait" (I'm thinking Rust) to reduce the number of modifications that we need to make to different profilers, to keep the same API

Reply:
I thought also about it, but then I thought 2 things:
 - I don't think it's quite related to this PR, as there are already arguments (like `insert_dso_name`) that aren't being used by all of the profilers
 - Profiling mode is something very major and all the profilers should get this argument

Reply:
Not this PR, okay